tools-local/doc.tasks/GenerateMarkdownDoc.cs

```diff
@@ -0,0 +1,127 @@
+    {
+        /// <summary>
+        /// Project files item group
+        /// </summary>
+        public ITaskItem[] ProductProjects { get; set; }
+        /// <summary>
+        /// Test project files item group
+        /// </summary>
+        public ITaskItem[] TestProjects { get; set; }
+        /// <summary>
+        /// Filepath to the generated documentation
+        /// </summary>
+        public ITaskItem OutputFile { get; set; }
+        /// <summary>
+        /// Repository Url to create links to the file
+        /// </summary>
+        public string GitHubRepositoryUrl { get; set; }
+        /// <summary>
+        /// Repository path on disk to relativize paths
+        /// </summary>
+        public string RepositoryRoot { get; set; }
+        /// <summary>
+        /// Title for the document
+        /// </summary>
+        public string Title { get; set; } = "NuGet Client Projects Overview";
+        /// <summary>
+        /// Sub-title for projects list
+        /// </summary>
+        public string ProjectsSubtitle { get; set; } = "Projects";
+        /// <summary>
+        /// Sub-title for test projects list
+        /// </summary>
+        public string TestProjectsSubtitle { get; set; } = "Test Projects";
+        /// <summary>
+        /// FullPath MSBuild item metadata
+        /// </summary>
+        private static readonly string MetaFullPath = "FullPath";
+
+        public override bool Execute()
+        {
+            var outputFilePath = OutputFile.GetMetadata(MetaFullPath);
+
+            var dirname = Path.GetDirectoryName(outputFilePath);
+            Directory.CreateDirectory(dirname);
+            Log.LogMessage(MessageImportance.Low, $"Created directories: {dirname}");
+
+            Log.LogMessage(MessageImportance.Low, $"Creating documentation: {outputFilePath}");
+            using (var file = new StreamWriter(outputFilePath))
+            {
+                file.WriteLine("---");
+                file.WriteLine($"date-generated: {DateTime.Now:s}");
+                file.WriteLine($"tool: {typeof(GenerateMarkdownDoc).FullName}");
+                file.WriteLine("---");
+
+                file.WriteLine($"\n\n# {Title}\n");
+                file.WriteLine("Below is a list of all source code projects for NuGet libraries and supported NuGet clients\n");
+
+                file.WriteLine($"\n\n## {ProjectsSubtitle}\n");
+                file.WriteLine("All shipped NuGet libraries and clients lives in `src/` folder.\n");
+                file.WriteLine($"Projects count: {ProductProjects.Length}\n");
+                SortAndWriteDescriptions(ProductProjects, file);
+
+                file.WriteLine($"\n\n## {TestProjectsSubtitle}\n");
+                file.WriteLine("Most production assemblies has an associated test project, whose name ends with `.Test`.\n");
+                file.WriteLine($"Test Projects count: {TestProjects.Length}\n");
+                SortAndWriteDescriptions(TestProjects, file);
```

Comment:
Shall we have a separate test-overview.md file for test projects?

Reply:
We can have a separte file. Now, the task supports generating two tasks. Please, review again.

Reply:
cc @donnie-msft thoughts on this. 
https://github.com/NuGet/Client.Engineering/issues/239 will be impacted by that decision.

Reply:
As we discussed offline, we can have a separate test-overview.md file.

Reply:
What is the rationale for having separate files?